### ISSUE TITLE
Crystal 0.35.1 has issue with WebSocket. Connection failed -> reserved1

### ISSUE THREAD

**Issue by msa7** [crystal-lang/crystal#9522]
I update to Crystal 0.35.1 via official docker image and got WebSocket issue. Rollback to 0.34.0 solve the issue.

```
web_socket.js:36 WebSocket connection to 'ws://localhost:8082/notifications?sid=74q9rbuX_u2Q' failed: One or more reserved bits are on: reserved1 = 0, reserved2 = 1, reserved3 = 0
setupSocket @ web_socket.js:36
(anonymous) @ web_socket.js:76
```

My WebSocket server is Kemal. Route setup

```
  ws "/notifications" do |socket, env|
    user_id_sockets[env.iam_id] ||= Array(HTTP::WebSocket).new
    user_id_sockets[env.iam_id] << socket

    socket.on_message do |message|
      if message == "ping"
        User.online(env.iam)
        socket.send("pong")
      end
    end

    socket.on_close do |_|
      sockets = user_id_sockets[env.iam_id]
      sockets.delete(socket)
    end
  end
```
 Related https://github.com/crystal-lang/crystal/issues/9484

**Comment by asterite** [comment]
Most likely one of these broke it: https://github.com/crystal-lang/crystal/pull/9313 https://github.com/crystal-lang/crystal/pull/8975

Can you try reverting those in the standard library and check if it works again?

**Comment by asterite** [comment]
Can you provide steps to reproduce? (how do you connect to Crystal?)

**Comment by xaviablaza** [comment]
@msa7 This is resolved in kemalcr/kemal#577
